core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -166,4 +179,36 @@ export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
   isLanguageRTL(): boolean {
     return this.i18nLanguageCodeService.isCurrentLanguageRTL();
   }
+
+  async saveLoggedOutProgress(): Promise<void> {
+    if (!this.loggedOutProgressUniqueUrlId) {
+      this.explorationPlayerStateService
+        .setUniqueProgressUrlId()
+        .then(() => {
+          this.loggedOutProgressUniqueUrlId = (
+            this.explorationPlayerStateService.getUniqueProgressUrlId());
+          this.loggedOutProgressUniqueUrl = (
+            this.urlService.getOrigin() +
+            '/progress/' + this.loggedOutProgressUniqueUrlId);
+        });
+    }
+    this.saveProgressMenuIsShown = true;
+  }
+
+  copyProgressUrl(): void {
+    navigator.clipboard.writeText(this.loggedOutProgressUniqueUrl);
```

Comment:
Can we use the Angular CDK clipboard instead? We seem to use it already in the `beam-job-output-dialog` component.

Reply:
Done!